### PR TITLE
QOI fixes

### PR DESCRIPTION
**Pull request by jmaselbas**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/ImageMagick/ImageMagick/pulls) open
- [x] I have verified that I am following the existing coding patterns and practices as demonstrated in the repository.

### Description
shortlog:
 coders: qoi: Fix unexpected end-of-file
 coders: qoi: Add the image end marker
 coders: qoi: Rework writer exception
 coders: qoi: Typo fix
 coders: qoi: Silence warnings
 coders: qoi: Close the image when writting is done
 coders: qoi: Fix indentation
 coders: qoi: Remove unused quantum_info

### CODE REVIEW

File: coders/qoi.c [hunk 1]
```diff
@@ -77,7 +77,12 @@
 #define QOI_COLOR_HASH(C) (C.rgba.r*3 + C.rgba.g*5 + C.rgba.b*7 + C.rgba.a*11)
 
 /*
-  Typedef declaractions.
+ Static Const declarations.
+*/
+static const unsigned char qoi_endmarker[8] = {0,0,0,0,0,0,0,1};
```
Comment (dlemstra) [inline coders/qoi.c:82]: Could we not include this in the method itself and write all the bytes at one?

File: coders/qoi.c [hunk 2]
```diff
@@ -325,8 +325,11 @@ static Image *ReadQOIImage(const ImageInfo *image_info,
       "UnexpectedEndOfFile",image->filename);
   if (status != MagickFalse) {
     status=SyncAuthenticPixels(image,exception);
-    if (image->ping == MagickFalse)
-      SyncImage(image,exception);
+    if (image->ping != MagickFalse) {
```
Comment (dlemstra) [inline coders/qoi.c:328]: This `if` block can be moved to after the line that has the following statement `ThrowReaderException(CorruptImageError,"NegativeOrZeroImageSize");`

File: coders/qoi.c (hunk 1 above)
Comment (jmaselbas) [reply coders/qoi.c:82]: using WriteBlobLongLong ?

Comment (dlemstra) [reply coders/qoi.c:82]: Probably easier with `WriteBlob` instead.

Comment (jmaselbas) [reply coders/qoi.c:82]: a `WriteBlobMSBLongLong(image, 0x1ull);` would have done the trick :) but there is no such  function I'll use WriteBlob instead

**Comment by jmaselbas** [comment]
In case you haven't noticed this has been updated, is anything that remains to be done ?

**Comment by dlemstra** [comment]
Haven't had time to take a look at this again. Looks good now.

**Pull request merged**